### PR TITLE
CMake: improve warning for udev rule installation

### PR DESCRIPTION
**Pull request by Be-ing**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -1294,10 +1294,21 @@ if(UNIX AND NOT APPLE)
...
-        "which is reserved for user-installed files. The correct directory for "
-        "system rules is /lib/udev/rules.d. with a appropiated priority prefix. "
-        "Adjust your package script accordingly and set -DINSTALL_USER_UDEV_RULES=OFF")
+    if(NOT CMAKE_INSTALL_SYSCONFDIR STREQUAL /etc OR /lib)
+        get_filename_component(UDEV_SOURCE_ABSOLUTE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/udev/rules.d" ABSOLUTE)
+        get_filename_component(UDEV_INSTALL_ABSOLUTE_PATH "${CMAKE_INSTALL_SYSCONFDIR}/udev/rules.d" ABSOLUTE)
+        message(WARNING
+"Installing udev rule file for USB HID and Bulk controller permissions to
+${UDEV_INSTALL_ABSOLUTE_PATH}
+If you are installing Mixxx from source for your own use, copy
```
Comment (daschuer) [inline CMakeLists.txt:1303]: The copying is not required, because it is done during
sudo make install

**Review by daschuer — CHANGES_REQUESTED** [review]

File: CMakeLists.txt (hunk 1 above)
Comment (Be-ing) [reply CMakeLists.txt:1303]: only if CMAKE_INSTALL_SYSCONFDIR is /etc or /lib. If the user is installing elsewhere, we should not assume they are running `make install` with root permissions

Comment (daschuer) [reply CMakeLists.txt:1303]: CMAKE_INSTALL_SYSCONFDIR is read only and either /etc or etc
The later etc case indicates that the user is installing into a temp directory most likely for packaging.
I don't see other use cases. 
Which use case do you refere here?

Comment (Be-ing) [reply CMakeLists.txt:1303]: I install to ~/local and the udev rule gets installed to /home/be/local/etc/udev/rules.d/mixxx-usb-uaccess.rules

File: CMakeLists.txt [hunk 2]
```diff
@@ -1294,10 +1294,21 @@ if(UNIX AND NOT APPLE)
       DESTINATION
         "${CMAKE_INSTALL_SYSCONFDIR}/udev/rules.d"
     )
-    message(WARNING "Installing udev rules to ${CMAKE_INSTALL_SYSCONFDIR}/udev/rules.d, "
-        "which is reserved for user-installed files. The correct directory for "
-        "system rules is /lib/udev/rules.d. with a appropiated priority prefix. "
-        "Adjust your package script accordingly and set -DINSTALL_USER_UDEV_RULES=OFF")
+    if(NOT CMAKE_INSTALL_SYSCONFDIR STREQUAL /etc OR /lib)
+        get_filename_component(UDEV_SOURCE_ABSOLUTE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/udev/rules.d" ABSOLUTE)
+        get_filename_component(UDEV_INSTALL_ABSOLUTE_PATH "${CMAKE_INSTALL_SYSCONFDIR}/udev/rules.d" ABSOLUTE)
```
Comment (Be-ing) [inline CMakeLists.txt:1299]: Huh this is printed as /home/be/sw/mixxx/etc/udev/rules.d but CMake actually installs it to /home/be/local/etc/udev/rules.d/mixxx-usb-uaccess.rules

**Pull request closed without merging**